pkg/k3s/Dockerfile

```diff
@@ -1,4 +1,4 @@
-ARG GOVER=1.14.4
```

Comment:
Not sure how we use pkg/k3s, but GOVER here looks like noop, we will use go bundled into eve-alpine regardless of this ARG.

Reply:
@giggsoff where is the version 1.16.15 set in the Dockerfiles you mentioned above ?

Reply:
Is that in pkg/pillar/go.mod ?

Reply:
We use apk add command (you can see the line in Dockerfiles in my links) to update go onto the version from another release of alpine (we use 3.13 as a base for eve-alpine, but it contains only go [1.15](https://pkgs.alpinelinux.org/package/v3.13/community/x86_64/go)).

Reply:
ok 3.16 seems to have latest go 1.18.5, I will update that.

Reply:
Running into build issues in containerd
#22 0.294 Running go vet                                                                                                                                                                       
#22 7.578 # github.com/containerd/containerd/mount                                                                                                                                             
#22 7.578 vendor/github.com/containerd/containerd/mount/temp.go:33:16: undefined: os.MkdirTemp                                                                                                 
#22 13.19 # github.com/containerd/containerd/content
#22 13.19 vendor/github.com/containerd/containerd/content/helpers.go:236:27: undefined: io.Discard
#22 14.43 # github.com/containerd/containerd/cio
#22 14.43 vendor/github.com/containerd/containerd/cio/io_unix.go:41:14: undefined: os.MkdirTemp
#22 15.81 # github.com/containerd/containerd/remotes/errors
#22 15.81 vendor/github.com/containerd/containerd/remotes/errors/errors.go:43:10: undefined: io.ReadAll
#22 20.26 # github.com/containerd/containerd/services/introspection
#22 20.26 vendor/github.com/containerd/containerd/services/introspection/local.go:117:15: undefined: os.ReadFile
#22 20.26 vendor/github.com/containerd/containerd/services/introspection/local.go:141:12: undefined: os.WriteFile

Investigating,

Reply:
@giggsoff do you have any idea ? Both with 3.15 and 3.16 alpine we see this error. 3.14 is fine.

Reply:
@giggsoff can we create an lfedge/eve-alpine which uses alpine 3.16 and then test if we can use it in the docker files?

Reply:
@zedi-pramodh what changes did you do to your workspace to get the above errors? I don't think MkdirTemp, ReadAll etc were removed from the go standard packages, so if I can run with your changes I can check what is going on. I don't know if go vet runs differently than the rest of the build. An experiment would be to comment it out from the Makefile and see.

But if you push your chanegs to the branch I can take a look; if not tonight I can look tomorrow morning.

Reply:
Change is very simple.

diff --git a/pkg/pillar/Dockerfile b/pkg/pillar/Dockerfile
index 2a37de3d8..700b49c58 100644
--- a/pkg/pillar/Dockerfile
+++ b/pkg/pillar/Dockerfile
@@ -10,7 +10,8 @@ RUN eve-alpine-deploy.sh
 
 # FIXME bump eve-alpine to alpine 3.14
 # hadolint ignore=DL3018
-RUN apk --no-cache --repository https://dl-cdn.alpinelinux.org/alpine/v3.14/community add -U --upgrade go && go version
+# bump alpine 3.16 to get go1.18.5
+RUN apk --no-cache --repository https://dl-cdn.alpinelinux.org/alpine/v3.16/community add -U --upgrade go && go version


Changed 3.14 to 3.16 which comes with go-1.18.5

Reply:
But this doesn't update lfedge/eve-alpine which is what we use is most of the build. Hence you end up with some old and some new golang.

I'll take a look tomorrow to see why it would complain about those functions being undefined.

Reply:
Updated commit with changes that is causing these build failures.

Reply:
@zedi-pramodh I can see during build with `RUN apk --no-cache --repository https://dl-cdn.alpinelinux.org/alpine/v3.16/community add -U --upgrade go && go version`:
```
Step 6/50 : RUN apk --no-cache --repository https://dl-cdn.alpinelinux.org/alpine/v3.16/community add -U --upgrade go && go version
 ---> Running in 194af4486bf6
fetch https://dl-cdn.alpinelinux.org/alpine/v3.16/community/x86_64/APKINDEX.tar.gz
WARNING: Ignoring https://dl-cdn.alpinelinux.org/alpine/v3.16/community: UNTRUSTED signature
OK: 565 MiB in 116 packages
go version go1.15.7 linux/amd64
```
So we do not update go at all and try to build with bundled go 1.15.7.
We can try to update `alpine-keys` in that case and slightly modify pillar to make it build-able under new go: [0001-Update-alpine-keys-to-get-go-from-3.16.txt](https://github.com/lf-edge/eve/files/9307454/0001-Update-alpine-keys-to-get-go-from-3.16.txt)

Answering the question about eve-alpine from @eriknordmark: yes, we can update eve-alpine to be based on top of 3.16, but this operation will update all packages and for example gcc we use to build kernel/qemu/anything else will be updated and may potentially come with new problems. So it will require additional testing on real hardware. Should I prepare an eve-alpine update?

Reply:
Thanks @giggsoff. So with the patch you provided I should be able to get the build going right ? I will give a try.

Reply:
The patch works. I could now compile eve with Golang 1.18.5. Amended the commit. We should run eden tests with this commit.

Reply:
@giggsoff I share @eriknordmark 's concerns. We should not be retrieving packages directly from the Internet for any of these packages other than eve-alpine.

If we need multiple versions of go or any other package, we should add them to eve-alpine in the mirror, and then use them downstream in pkg/edgeview or wherever

Reply:
@deitch seems I found the way to merge both alpine versions in eve-alpine: https://github.com/lf-edge/eve/pull/2762